src/react/components/GenericForm.jsx

```diff
@@ -18,112 +62,180 @@ const { TextArea } = Input;
-    return fields.map((field) => {
-        return (
-            <Form.Item
-                key={ field.key || field.name }
-                name={ field.name }
-                label={ formSettings.showLabels ? field.label : '' }
-                rules={ field.rules }
-            >
-                { createFormInput( field ) }
-            </Form.Item>
-        );
-    });
-}
+    /**
+     * Create single Form.Item content with provided field properties
+     * 
+     * @param {Object} field              - object containing information for single field
+     * @param {String} field.type         - field type as string {text / textarea / info / dropdown}
+     * @param {String} field.placeholder  - placeholder text for the current field
+     * @param {String|Number} field.value - value for current field used in info card / drowdown / submit button
+     * 
+     * @returns {Component} React component for the provided field
+     */
 
+    createFormInput (field) {
+        if (!field) {
+            return null;
+        }
+    
+        const fieldKey = field.name + '_' + field.type + '_field';
+    
+        switch(field.type) {
+            case 'text':
+                return (
+                    <Input
+                        key={ fieldKey }
+                        placeholder={ field.placeholder }
+                    />
+                );
+            case 'textarea':
+                return (
+                    <TextArea
+                        key={ fieldKey }
+                        placeholder={ field.placeholder }
+                    />
+                );
+            case 'info':
+                return (
+                    <Card key={ fieldKey }>
+                        { field.value }
+                    </Card>
+                );
+            case 'dropdown':
+                return (
+                    <Select
+                        key={ fieldKey }
+                        placeholder={ field.placeholder }
+                        dropdownStyle={{ zIndex: zIndexValue }}
 
-/**
- * Create single Form.Item content with provided field properties
- * 
- * @param {Object} field              - object containing information for single field
- * @param {String} field.type         - field type as string {text / textarea / info / dropdown}
- * @param {String} field.placeholder  - placeholder text for the current field
- * @param {String|Number} field.value - value for current field used in info card / drowdown / submit button
- * 
- * @returns {Component} React component for the provided field
- */
-const createFormInput = (field) => {
-    if (!field) {
-        return null;
+                    >
+                        { field.value.map((singleOption) => {
+                            return (
+                                <Select.Option
+                                    value={ singleOption.categoryId }
+                                    key={ singleOption.categoryId }
```

Comment:
`option.categoryId` doesn't make sense on a generic form. Use `option.value` instead and map it accordingly on myplaces functionality.

Reply:
Maybe even support non-object options? 
```suggestion
                                    key={ singleOption.value || singleOption }
```